src/main/java/com/julienviet/pgclient/impl/PostgresSQLConnection.java

```diff
@@ -142,4 +198,64 @@ public SQLConnection getTransactionIsolation(Handler<AsyncResult<TransactionIsol
     conn.schedule(new PreparedTxQueryCommand(handler));
     return this;
   }
+
+  @Override
+  public void close(Handler<AsyncResult<Void>> handler) {
+    inAutoCommit.set(true);
+    if (inTransaction.compareAndSet(true, false)) {
+      conn.schedule(new QueryCommand("COMMIT", new ResultSetBuilder(re -> {
+        if(re.failed()) {
+          handler.handle(Future.failedFuture(re.cause()));
+        } else {
+          // should we use close command for open portal if any ?
+          conn.doClose();
+          handler.handle(Future.succeededFuture());
+        }
+      })));
+    } else {
+      conn.doClose();
+      handler.handle(Future.succeededFuture());
+    }
+  }
+
+  @Override
+  public void close() {
+    close(c -> {
+    });
+  }
+
+  private void beginTxIfNeeded(Handler<AsyncResult<Void>> handler) {
+    if (!inAutoCommit.get() && inTransaction.compareAndSet(false, true)) {
+      conn.schedule(new QueryCommand("BEGIN", new ResultSetBuilder(re -> {
+        if(re.failed()) {
+          handler.handle(Future.failedFuture(re.cause()));
+        } else {
+          handler.handle(Future.succeededFuture());
+        }
+      })));
+    } else {
+      handler.handle(Future.succeededFuture());
+    }
+  }
+
+  private void endAndBeginTx(String command, Handler<AsyncResult<Void>> handler) {
+    if (inTransaction.compareAndSet(true, false)) {
+      conn.schedule(new QueryCommand(command, new ResultSetBuilder(re -> {
+        if(re.failed()) {
+          handler.handle(Future.failedFuture(re.cause()));
+        } else {
+          conn.schedule(new QueryCommand("BEGIN", new ResultSetBuilder(rs -> {
```

Comment:
IMHO, I prefer later because we need to implement other SQL connection methods with tests then we enhance